### PR TITLE
NN-3780: New endpoint to create drafts from reported adjudications

### PR DESCRIPTION
**Pull request by sp-watson**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/entities/DraftAdjudication.kt [hunk 1]
```diff
@@ -11,6 +11,7 @@ import javax.persistence.Table
 @Table(name = "draft_adjudications")
 data class DraftAdjudication(
   override val id: Long? = null,
+  val reportNumber: Long?,
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/entities/DraftAdjudication.kt:14]: This is required so we can filter out drafts that have already been reported

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/entities/IncidentDetails.kt [hunk 2]
```diff
@@ -6,7 +6,7 @@ import javax.persistence.Table
 
 @Entity
 @Table(name = "incident_details")
-class IncidentDetails(
+data class IncidentDetails(
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/entities/IncidentDetails.kt:9]: Not sure why this was missing, but it made it a lot harder to test, so added

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/gateways/ReportedAdjudication.kt [hunk 3]
```diff
@@ -10,6 +10,7 @@ data class ReportedAdjudication(
   val reporterStaffId: Long,
   val offenderNo: String,
   val bookingId: Long,
+  val agencyId: String,
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/gateways/ReportedAdjudication.kt:13]: We need this so we can return a draft with the agency id

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/controllers/ReportedAdjudicationController.kt [hunk 4]
```diff
@@ -50,4 +54,15 @@ class ReportedAdjudicationController {
   ): Page<ReportedAdjudicationDto> {
     return reportedAdjudicationService.getMyReportedAdjudications(agencyId, pageable)
   }
+
+  @PostMapping(value = ["/{adjudicationNumber}/create-draft-adjudication"])
```
Comment (richardpopple) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/controllers/ReportedAdjudicationController.kt:58]: Were we going to call them something other than drafts? Actually now I say it - I guess if they are actually drafts then that seems like a reasonable name

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/entities/DraftAdjudication.kt (hunk 1 above)
Comment (richardpopple) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/entities/DraftAdjudication.kt:14]: Cool

**Review by richardpopple — APPROVED** [review]

**Pull request merged**